lib/pbench/server/database/models/datasets.py

```diff
@@ -467,6 +473,31 @@ def is_tarball(path: Union[Path, str]) -> bool:
         """
         return str(path).endswith(Dataset.TARBALL_SUFFIX)
 
+    @staticmethod
+    def stem(path: Union[str, Path]) -> str:
+        """
+        The Path.stem() removes a single suffix, so our standard "a.tar.xz"
+        returns "a.tar" instead of "a". We could double-stem, but instead
+        this just checks for the expected 7 character suffix and strips it.
+
+        If the path does not end in ".tar.xz" then the full path.name is
+        returned.
+
+        Args:
+            path: A file path that might be a Pbench tarball
+
+        Raises:
+            BadFilename: the path name does not end in TARBALL_SUFFIX
```

Comment:
s/BadFilename/DatasetBadName/

Reply:
That's up to you; if you don't resolve the conversation soon, I guess I will, and I can rebase manually and such as well. I'd simply preferred not to wait for another round of approvals based on this minor issue.